Basic GoLang highlighting.

elliotcourant opened the pull request:
I've started adding some very basic GoLang highlighting. Please provide some feedback. 

![alt text](https://i.imgur.com/xNdeU98.png "Preview")

elliotcourant committed:
Trying to add Golang support.

elliotcourant committed:
Adding some basic highlighting for Golang.

kaicataldo commented on colors/material.vim:
```diff
@@ -189,6 +189,21 @@ call s:HL('jsTemplateExpression', s:red, '', '')
 
+" Golang 
+call s:HL('goComment', s:gray5, '', '')
+call s:HL('goFunctionCall', s:blue, '', 'bold')
```
I personally have a preference not to use bold text for syntax highlighting (it's used to highlight bold text in markdown files and for some Vim-specific features).

elliotcourant replied:
I'll tweak that!

kaicataldo reviewed (commented):
Hey, thanks for the PR! Do you mind sharing a screenshot of what it looks like with what's currently in master and with your changes?

kaicataldo commented on colors/material.vim:
```diff
@@ -189,6 +189,21 @@ call s:HL('jsTemplateExpression', s:red, '', '')
 call s:HL('jsUndefined', s:orange, '', '')
 
+" Golang 
+call s:HL('goComment', s:gray5, '', '')
```
Is there a reason [this line](https://github.com/kaicataldo/material.vim/pull/11/files#diff-693a3fb1845dd2c150138d65ec2fe8d6R101) doesn't cover comments in Golang? What Golang language plugin are you using?

Also, would you mind updating it to `s:gray4` so it matches?

elliotcourant replied:
I'm currently using `vim-go` but that highlighting is probably redundant. I'll remove it and see if it still works properly, if not I'll update it to `s:gray4`

elliotcourant commented:
I ended up tweaking the comment color to `gray4`, and I've removed the bold. I will post a screenshot of the new colors that I've added and the current colors without my changes here shortly. Thank for the feedback so far!

elliotcourant commented:
![image](https://user-images.githubusercontent.com/37967690/52575383-b80a0000-2de3-11e9-85ad-635886adb02e.png)

This is how it looks without my changes. My changes were trying to match https://plugins.jetbrains.com/plugin/8006-material-theme-ui as closely as I could.

kaicataldo commented:
Thanks for contributing!
